velox/common/memory/Allocation.cpp

```diff
@@ -32,24 +32,10 @@ Allocation::~Allocation() {
 
 void Allocation::append(uint8_t* address, int32_t numPages) {
   numPages_ += numPages;
-  if (runs_.empty()) {
-    runs_.emplace_back(address, numPages);
-    return;
-  }
-
-  PageRun last = runs_.back();
-  VELOX_CHECK_NE(
-      address, last.data(), "Appending a duplicate address into a PageRun");
-
-  // Increment page count if new data starts at end of the last run
-  // and the combined page count is within limits.
-  if ((address ==
-       last.data() + last.numPages() * AllocationTraits::kPageSize) &&
-      (last.numPages() + numPages <= PageRun::kMaxPagesInRun)) {
-    runs_.back() = PageRun(last.data(), last.numPages() + numPages);
-  } else {
-    runs_.emplace_back(address, numPages);
-  }
+  VELOX_CHECK(
+      runs_.empty() || address != runs_.back().data(),
```

Comment:
I'm curious why do we choose to check address against the start of the last run. Shouldn't we check that it doesn't match the start of any run in 'runs_'?

Reply:
That's a good point. This is a best effort check for now. Not sure if we were concerning about the check cost of an allocation with many page runs. Ideally we shall check if there is any overlap in buffer spaces. I will leave a TODO there.